### ISSUE TITLE
[Structural Subtyping] Error when implementing a property + classmethod and checking it against a Prototype

### ISSUE THREAD

**Issue by gustavocmv** [microsoft/pyright#3869]
**Describe the bug**
Tried to define a `Protocol` to enforce a "class property" (a method decorated with both `classmethod` and `property`) with a given output type. When type-checking a class that implements that "class property", VS Code's Pylance accused a problem.

**To Reproduce**
```py
from typing import Protocol


class MyProtocol(Protocol):
    @classmethod
    @property
    def my_class_property(cls) -> str:
        ...


class MyClass:
    @classmethod
    @property
    def my_class_property(cls) -> str:
        return "my_prop_value"


cls: MyProtocol = MyClass
```
The last line results in the following Problem from Pylance in VS Code:
```
Expression of type "Type[MyClass]" cannot be assigned to declared type "MyProtocol"
  "my_class_property" is an incompatible type
    "property" is incompatible with "str"
```

**Expected behavior**
As far as I understand, this implementation is right, i.e, Pylance should not mark it as a Problem.
However, I am still learning about python typing, so please let me know if my example is wrong and how I can fix it.

**VS Code extension or command-line**
- Running on VS Code -- `vscode --version` output:
```
1.70.2
e4503b30fc78200f846c62cf8091b76ff5547662
x64
```
- Pylance Extension version: `v2022.8.40`

**Comment by erictraut** [comment]
You need to declare the type of `cls` to be `type[MyProtocol]`.

```python
cls: type[MyProtocol] = MyClass
```

**Comment by gustavocmv** [comment]
@erictraut , thanks for the correction! I used it and the error is gone!

However, when trying to use `MyProtocol` to type-hint a function `fun` and using the defined class-property, I got the error:
```
Cannot access member "my_class_property" for type "Type[MyProtocol]"
  A property defined within a protocol class cannot be accessed as a class variable
```
Is it working as designed? How can I type-hint this use-case?

Once again, thanks for the help!


**To Reproduce**
```py
# Using previous code...

cls: type[MyProtocol] = MyClass  # Works

print(cls.my_class_property)  # Works

def fun(cls: type[MyProtocol]):
    return cls.my_class_property  # Error
```

**Comment by erictraut** [comment]
Support for class properties was nominally added in Python 3.8, but the concept was not well conceived and caused many problems (runtime bugs, undefined and unexpected behaviors, etc.). Support for class properties has been been removed in 3.11. I recommend not using them.

**Comment by gustavocmv** [comment]
@erictraut thanks again!
I will rethink my design to avoid them.
